packages/fxa-auth-server/lib/payments/configuration/plan.ts

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import joi from '@hapi/joi';
+
+import {
+  BaseConfig,
+  baseConfigSchema,
+  CapabilityConfig,
+  StyleConfig,
+  SupportConfig,
+  UiContentConfig,
+  UrlConfig,
+} from './base';
+
+export const planConfigSchema = baseConfigSchema.keys({
+  stripePriceId: joi.string().optional(),
+  productOrder: joi.number().optional(),
+  googlePlaySku: joi.array().items(joi.string()).optional(),
+  appleProductId: joi.array().items(joi.string()).optional(),
+});
+
+export class PlanConfig implements BaseConfig {
+  // Firestore document id
+  id: string;
+  // Firestore ProductConfig document id
+  productConfigId: string;
+
+  active: boolean;
+  capabilities?: CapabilityConfig;
+  urls?: UrlConfig;
+  uiContent?: UiContentConfig;
+  styles?: StyleConfig;
+  locales?: {
+    [key: string]: {
+      uiContent: Partial<UiContentConfig>;
+      urls: Partial<UrlConfig>;
+      support: Partial<SupportConfig>;
+    };
+  };
+  support?: SupportConfig;
+
+  // Extended by PlanConfig
+  stripePriceId?: string;
+  productOrder?: number;
+  googlePlaySku?: string[];
+  appleProductId?: string[];
```

Comment:
Kind of a Stripe Plan specific information, but our [existing Joi plan validator](https://github.com/mozilla/fxa/blob/f32ae3f1a02969766a7ba4686907f941793a8fa6/packages/fxa-auth-server/lib/routes/validators.js#L502-L514) currently requires some additional keys:
* `interval`
* `interval_count`
* `amount`
* `currency`

These could all be fetched if we had a Stripe plan ID. Do we want to make them optional? @chenba

Reply:
I don't understand why we'd want to replicate Stripe data here. Just like there's no validation of google or apple specific receipt/token data. The google/apple/stripe correlated plans could all have different prices. If a stripe price is linked, we could certainly show its information in the UX when viewing this though.